integration/test/cloudwatchlogs/publish_logs_test.go

```diff
@@ -77,6 +80,44 @@ func TestWriteLogsToCloudWatch(t *testing.T) {
 	}
 }
 
+// TestRotatingLogsDoesNotSkipLines validates https://github.com/aws/amazon-cloudwatch-agent/issues/447
+// The following should happen in the test:
+// 1. A log line of size N should be written
+// 2. The file should be rotated, and a new log line of size N should be written
+// 3. The file should be rotated again, and a new log line of size GREATER THAN N should be written
+// 4. All three log lines, in full, should be visible in CloudWatch Logs
+func TestRotatingLogsDoesNotSkipLines(t *testing.T) {
+	cfgFilePath := "resources/config_log_rotated.json"
+
+	instanceId := test.GetInstanceId()
+	log.Printf("Found instance id %s", instanceId)
+	logGroup := instanceId
+	logStream := instanceId + "Rotated"
+
+	defer test.DeleteLogGroupAndStream(logGroup, logStream)
+
+	start := time.Now()
+	test.CopyFile(cfgFilePath, configOutputPath)
+
+	test.StartAgent(configOutputPath)
+
+	// ensure that there is enough time from the "start" time and the first log line,
+	// so we don't miss it in the GetLogEvents call
+	time.Sleep(agentRuntime)
+	t.Log("Writing logs and rotating")
+	// execute the script used in the repro case
+	test.RunCommand("/usr/bin/python3 resources/write_and_rotate_logs.py")
+	time.Sleep(agentRuntime)
+	test.StopAgent()
+
+	lines := []string{
```

Comment:
Would be more good to add the comments for these validated logs line. (Based on the order and logs' content of https://github.com/aws/amazon-cloudwatch-agent/pull/457/files#diff-dbb09475b756bd0a76e4d46daaae98ec5f3668356ecdb24885493c770ea31be8R20-R28)

Reply:
Will add comments